server/app.ts

```diff
@@ -0,0 +1,195 @@
+import indexRoutes from './routes'
+import healthcheck from './services/healthCheck'
+import nunjucksSetup from './utils/nunjucksSetup'
+import config from './config'
+import errorHandler from './errorHandler'
+import standardRouter from './routes/standardRouter'
+import authorisationMiddleware from './middleware/authorisationMiddleware'
+import type UserService from './services/userService'
+
+const version = Date.now().toString()
+const production = process.env.NODE_ENV === 'production'
+const testMode = process.env.NODE_ENV === 'test'
+const RedisStore = connectRedis(session)
+
+export default function createApp(userService: UserService): express.Application {
+  const app = express()
+
+  auth.init()
+
+  app.set('json spaces', 2)
+
+  // Configure Express for running behind proxies
+  // https://expressjs.com/en/guide/behind-proxies.html
+  app.set('trust proxy', true)
+
+  // View Engine Configuration
+  app.set('view engine', 'njk')
+
+  nunjucksSetup(app, path)
+
+  // Server Configuration
+  app.set('port', process.env.PORT || 3000)
+
+  // Secure code best practice - see:
+  // 1. https://expressjs.com/en/advanced/best-practice-security.html,
+  // 2. https://www.npmjs.com/package/helmet
+  app.use(
+    helmet({
+      contentSecurityPolicy: {
+        directives: {
+          defaultSrc: ["'self'"],
+          // Hash allows inline script pulled in from https://github.com/alphagov/govuk-frontend/blob/master/src/govuk/template.njk
+          scriptSrc: ["'self'", 'code.jquery.com', "'sha256-+6WnXIl4mbFTCARd8N3COQmT3bJJmo32N8q8ZSQAIcU='"],
+          styleSrc: ["'self'", 'code.jquery.com'],
+          fontSrc: ["'self'"],
+        },
+      },
+    })
+  )
+
+  app.use(addRequestId())
+
+  const client = redis.createClient({
+    port: config.redis.port,
+    password: config.redis.password,
+    host: config.redis.host,
+    tls: config.redis.tls_enabled === 'true' ? {} : false,
+  })
+
+  app.use(
+    session({
+      store: new RedisStore({ client }),
+      cookie: { secure: config.https, sameSite: 'lax', maxAge: config.session.expiryMinutes * 60 * 1000 },
+      secret: config.session.secret,
+      resave: false, // redis implements touch so shouldn't need this
+      saveUninitialized: false,
+      rolling: true,
+    })
+  )
+
+  app.use(passport.initialize())
+  app.use(passport.session())
+
+  // Request Processing Configuration
+  app.use(bodyParser.json())
+  app.use(bodyParser.urlencoded({ extended: true }))
+
+  app.use(flash())
+
+  // Resource Delivery Configuration
+  app.use(compression())
+
+  // Cachebusting version string
+  if (production) {
+    // Version only changes on reboot
+    app.locals.version = version
+  } else {
+    // Version changes every request
+    app.use((req, res, next) => {
+      res.locals.version = Date.now().toString()
+      return next()
+    })
+  }
+
+  //  Static Resources Configuration
+  const cacheControl = { maxAge: config.staticResourceCacheDuration * 1000 }
```

Comment:
Really don't like how this registration of middleware in our apps just sprawls

Think extracting it into separate chunks like they do in prison staff hub might be nice.. I was at first quite sceptical but quite like it now.  (alternative suggestions also welcome!)